rust/src/tree.rs

```diff
@@ -0,0 +1,214 @@
+#[derive(Clone)]
+pub struct Node {
+    pub children: Vec<Node>,
```

Comment:
Use dict of [first char] -> Nod, rather than vector.

Reply:
Made the change. great suggestion